source/application/transforms/rewrite-includes/index.js

```diff
@@ -0,0 +1,46 @@
+import {relative, join, basename} from 'path';
+
+export default function createWriteIncludesTransform (application, transformConfiguration) {
+	return async function rewriteIncludesTransform (file, demo, configuration) {
+		const patternConfig = application.configuration.patterns;
+		const resultName = patternConfig.formats[configuration.outFormat].name;
+
+		const source = file.buffer.toString('utf-8');
+		const resolvePath = configuration.resolve;
+
+		const IDRegistry = Object.keys(file.dependencies).reduce((registry, dependencyName) => {
```

Comment:
Its the same code as in the rewrite-imports transform. This should be kept dry. Create transform-utils as base for this. It could be a separate npm package later on for transform implementors.